app/registries/models.py

```diff
@@ -485,17 +494,20 @@ class RegistriesApplication(AuditModel):
         on_delete=models.PROTECT,
         verbose_name="Certificate")
     primary_certificate_no = models.CharField(max_length=50)
-
-    @property
-    def current_status(self):
-        try:
-            return RegistriesApplicationStatus.objects.get(
-                application=self.application_guid,
-                expired_date=None)
-        except:
-            logger.error('Could not find the current status for application: {}'.format(
-                self.application_guid))
-            return None
+    # TODO Should probably force this to have a default value of Pending!
+    current_status = models.ForeignKey(
+        ApplicationStatusCode,
+        blank=True,
+        null=True,
+        db_column='registries_application_status_code',
+        on_delete=models.PROTECT,
+        verbose_name="Application Status Code Reference")
```

Comment:
It would be good to look into what constraints we can place on the "relationship" between current_status and application_outcome_date

Reply:
Agreed. The data in the database however isn't very consistent. For example, most approved applications never had an application received date.

Reply:
"most approved applications never had an application received date." => "most *legacy* approved applications never had an application received date."

New world will be different.. EVERY application will have a received date.